src/SudoSu.php

```diff
@@ -23,15 +23,27 @@ public function __construct(Application $app, AuthManager $auth, SessionManager
         $this->session = $session;
     }
 
-    public function loginAsUser($userId, $currentUserId)
+    /**
+     * Log the sudosu user in and preserve the original user id if exists
+     *
+     * @param string $sudosuUserId
+     * @param string|null $originalUserId
+     * @return void
+     */
+    public function loginAsUser($sudosuUserId, $originalUserId)
     {
         $this->session->put('sudosu.has_sudoed', true);
-        $this->session->put($this->sessionKey, $currentUserId);
+        $this->session->put($this->sessionKey, $originalUserId);
 
-        $this->auth->loginUsingId($userId);
+        $this->auth->loginUsingId($sudosuUserId);
```

Comment:
I don't see the need to name it `$sudosuUserId` considering that `$userId` makes perfect sense in the given context. 

Currently the I don't think the code has any way in which it could be misinterpreted.
```php
$this->session->put('sudosu.has_sudoed', true);
$this->session->put($this->sessionKey, $originalUserId);

$this->auth->loginUsingId($userId);
```

I think `$userId` seems explicit enough.